homeassistant/components/foscam/coordinator.py

```diff
@@ -54,4 +54,20 @@ async def _async_update_data(self) -> dict[str, Any]:
                 self.session.is_asleep
             )
             data["is_asleep"] = {"supported": ret == 0, "status": is_asleep}
+
+            ret, is_openIr = await self.hass.async_add_executor_job(
+                self.session.get_infra_led_config
+            )
+            data["is_openIr"] = is_openIr["mode"]
+
+            ret, is_Flip = await self.hass.async_add_executor_job(
+                self.session.get_mirror_and_flip_setting
+            )
+            data["is_Flip"] = is_Flip["isFlip"]
+
+            ret, is_Flip = await self.hass.async_add_executor_job(
+                self.session.get_mirror_and_flip_setting
+            )
+            data["is_Mirror"] = is_Flip["isMirror"]
```

Comment:
Switching from the event loop to an executor job is quite expensive, so ideally we run this all in one executor job, so I would propose to refactor this code